qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
+you like, along with a collection of non-gate operations such as qubit measurements and
+reset operations.  However, when running a circuit on a real quantum device one no longer
+has this flexibility.  Due to limitations in, for example, the physical interactions
+between qubits, difficulty in implementing multi-qubit gates, control electronics etc,
+a quantum computing device can only natively support a handful of quantum gates and non-gate
+operations. The allowed instructions for a given backend can be found by querying the
+:class:`~.Target` for the devices:
 
-.. _transpiler_supplemental:
+.. code-block:
 
-Supplementary Information
-=========================
+   from qiskit.providers.fake_provider import FakeVigoV2
+   backend = FakeVigoV2()
 
-.. dropdown:: Basis gates
-   :animate: fade-in-slide-down
+   print(backend.target)
 
-   When writing a quantum circuit you are free to use any quantum gate (unitary operator) that
-   you like, along with a collection of non-gate operations such as qubit measurements and
-   reset operations.  However, when running a circuit on a real quantum device one no longer
-   has this flexibility.  Due to limitations in, for example, the physical interactions
-   between qubits, difficulty in implementing multi-qubit gates, control electronics etc,
-   a quantum computing device can only natively support a handful of quantum gates and non-gate
-   operations. In the present case of IBM Q devices, the native gate set can be found by querying
-   the devices themselves, and looking for the corresponding attribute in their configuration:
+Every quantum circuit run on the target device must be expressed using only these instructions.
+For example, suppose one wants to run a simple phase estimation circuit:
 
-   .. code-block::
+.. plot::
+   :include-source:
 
-      from qiskit.providers.fake_provider import FakeVigoV2
-      backend = FakeVigoV2()
+   import numpy as np
+   from qiskit import QuantumCircuit
+   from qiskit.providers.fake_provider import FakeVigoV2
 
-      print(backend.operation_names)
+   backend = FakeVigoV2()
 
-   .. parsed-literal::
+   qc = QuantumCircuit(2, 1)
 
-      ['id', 'rz', 'sx', 'x', 'cx', 'measure', 'delay']
+   qc.h(0)
+   qc.x(1)
+   qc.cp(np.pi/4, 0, 1)
+   qc.h(0)
+   qc.measure([0], [0])
+   qc.draw(output='mpl')
 
-   Every quantum circuit run on an IBM Q device must be expressed using only these basis gates.
-   For example, suppose one wants to run a simple phase estimation circuit:
+We have :math:`H`, :math:`X`, and controlled-:math:`P` gates, all of which are
+not in our devices basis gate set, and must be expanded.  This expansion is taken
+care of for us in the :func:`qiskit.execute` function. However, we can
+decompose the circuit to show what it would look like in the native gate set of
+the IBM Quantum devices (the :class:`~.FakeVigoV2` backend is a fake backend that
+models the historical IBM Vigo 5 qubit device for test purposes):
```

Comment:
```suggestion
We have :math:`H`, :math:`X`, and controlled-:math:`P` gates, none of which are
in our device's basis gate set, and thus must be expanded.  This expansion is taken
care of for us in the :func:`qiskit.execute` function. However, we can
decompose the circuit to show what it would look like in the native gate set of
the IBM Quantum device (the :class:`~.FakeVigoV2` backend is a fake backend that
models the historical IBM Vigo 5 qubit device for test purposes):
```